### ISSUE TITLE
CCK2 JavaScript Error when opening Firefox

### ISSUE THREAD

**Issue by CVR-EDV** [mkaply/cck2wizard#67]
Since the update on Quantum we got two errors. But only one of them on each system. This seems kind of random.

**First error:**
When opening Firefox we got this certain error.
![cck2_error](https://user-images.githubusercontent.com/33828309/33018749-e8161a92-cdf7-11e7-95ea-70a8c64e618e.PNG)
Is there something we can do about this?

**Second error:**
Furthermore we got the error “Failed to read the configuration file. Please contact your system administrator.”. You already have described this on your blog. Is it correct, that updating the modules folder will resolve the problem? I downloaded the latest cck2 release from GitHub and overwrited the modules folder with the new files. I still get the error. What am I doing wrong?

We are thinking of switching to the ESR 52.x version...

**Comment by mkaply** [comment]
So the first error should definitely be solved by overwriting the files in the modules folder.

The second problem is happening for some people and not others. They've been able to fix it by removing the startupCache directory in 

C:\Users\USERNAME\AppData\Local\Mozilla\Firefox\Profiles\profileName

I'm still trying to figure out what's happening.

**Comment by CVR-EDV** [comment]
I double checked if I really overwrited the old folder. To be sure I deleted the old and copied the GitHub release - still the same.

Secondly at a (local) PC where the the second error happens, I deleted the startupCache folder of the certain user. Still the same error. Firefox creted a new startupChache folder immediatly after that.

**Comment by mkaply** [comment]
Would you be willing to send me your cck2.cfg for testing?

Alternatively, could you add try/catch around these two lines in cck2.cfg and see if you can report the error in the Javascript console:

try {
Components.utils.import("resource://cck2/CCK2.jsm");
CCK2.init(config, "ä"[0], "ä");
} catch (e) {
Components.utils.reportError(e);
}

If you do this, Firefox should start and you will see the error in the Javascript console (Ctrl+Shift+J). This should help me figure out what is going on.

Thanks for any help you can provide.

**Comment by CVR-EDV** [comment]
I sent you an E-Mail with further info.

**Comment by care2bfree** [comment]
Curious, I'm having some similar cck2 problems so went to try the try/catch you mentioned above. The second line in my cck2.cfg is CCK2.init(config);. Just wondering if I'm missing something - should I have the same line you are showing above? If so, it looks like I'm seeing ascii character 132 there between the quotes - is that correct?

**Comment by care2bfree** [comment]
Curious, I'm having some similar cck2 problems so went to try the try/catch you mentioned above. My second line is CCK2.init(config);. Just wondering if I'm missing something - should I have the same line you are showing above? If so, it looks like I'm seeing ascii character 132 there between the quotes - is that correct?

**Comment by CVR-EDV** [comment]
It seems like we solved the error. 
My colleague downloaded the latest cck2 release (the wizard version I guess) from Mike's website and generated a new config. Then he generated a new group policy to publish the files. Then we got the "second error". Then I tried to delete the StartupCache folder and it worked! So we additionally created a policy to delete the StartupCache folder on all user accounts. Now everything works fine...

@Mike:
If you want any files from me, to find out what happened in detail, I can provide those if you wish.
